### PR TITLE
Agent: Added loading component to schedule list

### PR DESCRIPTION
**Pull request by abist**
Fixes https://github.com/microsoft/azuredatastudio/issues/5252

There wasn't an issue with creating job schedules, but it was slow if the server had a lot of schedules. Added a loading component with a spinning wheel while the schedules get populated so that it's apparent in the UI.

### CODE REVIEW

File: extensions/agent/src/data/pickScheduleData.ts [hunk 1]
```diff
@@ -22,10 +22,12 @@ export class PickScheduleData implements IAgentDialogData {
 
 	public async initialize() {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
-		}
+		await agentService.getJobSchedules(this.ownerUri).then((result) => {
+			if (result && result.success) {
```
Comment (udeeshagautam) [inline extensions/agent/src/data/pickScheduleData.ts]: just for my understanding: you want to check for result success or result schedules here?

Comment (abist) [reply extensions/agent/src/data/pickScheduleData.ts]: We want to look for success. The schedules can be an empty list.

File: extensions/agent/src/data/pickScheduleData.ts [hunk 2]
```diff
@@ -22,10 +22,12 @@ export class PickScheduleData implements IAgentDialogData {
 
 	public async initialize() {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
-		}
+		await agentService.getJobSchedules(this.ownerUri).then((result) => {
```
Comment (Charles-Gagnon) [inline extensions/agent/src/data/pickScheduleData.ts]: Why did you refactor this? We shouldn't be using .then if possible. And why are you returning true? You don't actually use the return value so why bother returning anything?

And as a side note - always make sure to update the return type for the function if you are adding one.

File: extensions/agent/src/dialogs/pickScheduleDialog.ts [hunk 3]
```diff
@@ -38,7 +39,17 @@ export class PickScheduleDialog {
 	}
 
 	public async showDialog() {
-		await this.model.initialize();
+		this.model.initialize().then((result) => {
```
Comment (Charles-Gagnon) [inline extensions/agent/src/dialogs/pickScheduleDialog.ts]: Again - please don't use .then. Just add the code inside it below the await this.model.initialize() call and it'll do the same thing.

File: extensions/agent/src/data/pickScheduleData.ts (hunk 1 above)
Comment (Charles-Gagnon) [inline extensions/agent/src/data/pickScheduleData.ts]: If the result isn't a success we should probably be displaying an error to the user

File: extensions/agent/src/data/pickScheduleData.ts (hunk 2 above)
Comment (abist) [reply extensions/agent/src/data/pickScheduleData.ts]: Fixes

File: extensions/agent/src/dialogs/pickScheduleDialog.ts (hunk 3 above)
Comment (abist) [reply extensions/agent/src/dialogs/pickScheduleDialog.ts]: This is intended. `.then()` works differently compared to `await`. We don't want to wait and block the UI for a backend job. This is what the PR is fixing.

File: extensions/agent/src/data/pickScheduleData.ts (hunk 1 above)
Comment (abist) [reply extensions/agent/src/data/pickScheduleData.ts]: Fixed

File: extensions/agent/src/dialogs/pickScheduleDialog.ts (hunk 3 above)
Comment (Charles-Gagnon) [reply extensions/agent/src/dialogs/pickScheduleDialog.ts]: Ah, yeah ok. This method doesn't have to be async then it looks like

File: extensions/agent/src/data/pickScheduleData.ts [hunk 4]
```diff
@@ -20,12 +20,18 @@ export class PickScheduleData implements IAgentDialogData {
...
+	public async initialize(): Promise<azdata.AgentJobScheduleInfo[]> {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
+		try {
+			let result = await agentService.getJobSchedules(this.ownerUri);
+			if (result && result.success) {
+				this.schedules = result.schedules;
+				return Promise.resolve(this.schedules);
```
Comment (Charles-Gagnon) [inline extensions/agent/src/data/pickScheduleData.ts]: You don't need to Promise.resolve here - just return this.schedules.

File: extensions/agent/src/data/pickScheduleData.ts [hunk 5]
```diff
@@ -20,12 +20,18 @@ export class PickScheduleData implements IAgentDialogData {
...
 
-	public async initialize() {
+	public async initialize(): Promise<azdata.AgentJobScheduleInfo[]> {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
+		try {
+			let result = await agentService.getJobSchedules(this.ownerUri);
+			if (result && result.success) {
```
Comment (Charles-Gagnon) [inline extensions/agent/src/data/pickScheduleData.ts]: >result [](start = 7, length = 6)

If the result isn't a success we should throw an error

File: extensions/agent/src/data/pickScheduleData.ts [hunk 6]
```diff
@@ -20,12 +20,18 @@ export class PickScheduleData implements IAgentDialogData {
...
 	}
 
-	public async initialize() {
+	public async initialize(): Promise<azdata.AgentJobScheduleInfo[]> {
 		let agentService = await AgentUtils.getAgentService();
-		let result = await agentService.getJobSchedules(this.ownerUri);
-		if (result && result.success) {
-			this.schedules = result.schedules;
+		try {
+			let result = await agentService.getJobSchedules(this.ownerUri);
```
Comment (Charles-Gagnon) [inline extensions/agent/src/data/pickScheduleData.ts]: I wouldn't bother wrapping this in a try/catch - it's just going to hide the original source of the error if an exception is thrown

**Pull request merged**